lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1187,3 +1197,23 @@ instance LiftHandler (Request, ServantErr) where
                 , renderHeader $ contentType $ Proxy @JSON
                 ) : headers
             }
+
+instance LiftHandler ErrListStakePools where
+    handler = \case
+        ErrListStakePoolsMetricsIsUnsynced ->
+            apiError err400 NotSynced $ mconcat
+                [ "I can't list stake pools yet because I need to scan the "
+                , "blockchain for metrics first."
+                ]
+        ErrListStakePoolsStakeIsUnreachable ->
+            apiError err400 StakeIsUnreachable $ mconcat
+                [ "I can't reach the stake distribution from jörmungandr. "
+                , "Maybe the jörmungandr is not started, or it is on a BFT "
+                , "blockchain instead of, Praos/Genesis?"
+                ]
+        ErrListStakePoolsInternalError _ ->
+            apiError err500 StakeIsUnreachable $ mconcat
+                [ "Internal error."
+                ]
```

Comment:
~StakeIsUnreachable~ --> `UnexpectedError`. 

Also, "Internal error" doesn't tell us anything about what could have happened. What's the case where we would get a 500 here?